src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int arrayIndexNode = getTableIndex(key);
+        Node<K, V> nodeByIndex = table[arrayIndexNode];
+        if (nodeByIndex == null) {
+            table[arrayIndexNode] = new Node<>(key, value, null);
+        } else {
+            while (nodeByIndex != null) {
+                if (Objects.equals(nodeByIndex.key, key)) {
+                    nodeByIndex.value = value;
+                    return;
+                }
+                if (nodeByIndex.next == null) {
+                    nodeByIndex.next = new Node<>(key, value,null);
+                    break;
+                }
+                nodeByIndex = nodeByIndex.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        int arrayIndexNode = getTableIndex(key);
+        Node<K, V> tempNode = table[arrayIndexNode];
```

Comment:
do not complicate names when there is no reason for it. Ту в методе только один индекс и одна нода, мы не спутаем эти переменные с чем-то еще, так что можно называть их просто.
```suggestion
        int index = getTableIndex(key);
        Node<K, V> node = table[index];
```